buttercup/cogs/heatmap.py

```diff
@@ -0,0 +1,258 @@
+import io
+from typing import Dict, List, Optional
+
+import matplotlib.pyplot as plt
+from blossom_wrapper import BlossomAPI
+from discord import File
+from discord.ext.commands import Cog
+from discord_slash import SlashContext, cog_ext
+from discord_slash.utils.manage_commands import create_option
+from matplotlib.font_manager import FontProperties
+
+from buttercup.bot import ButtercupBot
+
+
+# Colors to use in the plots
+background_color = "#36393f"  # Discord background color
+text_color = "white"
+line_color = "white"
+intensity_colors = [
+    "#07102c",
+    "#0a1842",
+    "#0e2058",
+    "#11286e",
+    "#153184",
+    "#18399a",
+    "#1c41b0",
+    "#1f49c7",
+    "#2251dd",
+    "#3862e0",
+]
+
+
+Heatmap = Dict[int, Dict[int, int]]
+
+
+def init_heatmap_dict() -> Heatmap:
+    """Get the initial dictionary for the heatmap data.
+
+    For every day (1 = Monday, 7 = Sunday), it contains a dict entry
+    with an entry 0 for every hour.
+    """
+    heatmap_dict = {}
+
+    for day in range(1, 8):
+        day_dict = {}
+
+        for hour in range(0, 24):
+            day_dict[hour] = 0
+
+        heatmap_dict[day] = day_dict
+
+    return heatmap_dict
+
+
+def get_heatmap_dict(data: List[Dict[str, int]]) -> Heatmap:
+    """Populate the heatmap dictionary with the data returned by Blossom."""
+    heatmap_dict = init_heatmap_dict()

```

Comment:
I'd recommend including `numpy` and `pandas` for data-oriented commands, This greatly simplifies the code in this PR. I'll provide examples to transform the code using these two libraries. Note that I am not at all sure that the code works, so be sure to check.

```
data_frame = pd.DataFrame(data)
heatmap = data_frame.groupBy(by=["day", "hour"]).sum()
```
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.groupby.html